// ai-vue-i18n/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var path = require('path');
var scriptLoader = require('./script-loader-D3WzPCA-.cjs');
require('assert');
require('module');
require('events');
require('node:fs');
require('node:fs/promises');
require('node:os');
require('node:path');
require('node:url');
require('node:worker_threads');
require('os');
require('node:util');
require('crypto');
require('fs');
require('util');
require('node:module');
require('node:assert');
require('url');
require('tty');

function vueI18nPlugin() {
    return {
        name: "vue-i18n-plugin-ai",
        enforce: "pre",
        async transform(code, filePath) {
            try {
                const commonOptions = await scriptLoader.fileOperator.getConfig();
                if (!scriptLoader.validateFileType(filePath, commonOptions)) {
                    return null;
                }
                const resourcePath = path.normalize(filePath);
                const localeFile = path.normalize(path.join(commonOptions.outputDir, `${commonOptions.anchorName}.js`));
                await scriptLoader.fileOperator.initMessage(localeFile);
                if (!scriptLoader.fileOperator.messages) {
                    return { code };
                }
                // 处理js文件
                if ([".js", ".ts", ".cjs"].includes(path.extname(resourcePath))) {
                    const scriptLoader$1 = new scriptLoader.ScriptLoader(commonOptions, resourcePath);
                    code = scriptLoader$1.excute(code, false);
                    return { code };
                }
                // 处理vue文件
                else if (filePath.endsWith(".vue")) {
                    const { template, script, scriptSetup } = scriptLoader.getVueModule(code, commonOptions.vueVersion);
                    // 处理template
                    if (template) {
                        const templateLoader = new scriptLoader.TemplateLoader(commonOptions, resourcePath);
                        const result = templateLoader.excute(template);
                        code = code.replace(template, result);
                    }
                    // 处理script
                    if (script) {
                        const scriptLoader$1 = new scriptLoader.ScriptLoader(commonOptions, resourcePath);
                        const result = scriptLoader$1.excute(script, false);
                        code = code.replace(script, result);
                    }
                    if (scriptSetup) {
                        const scriptLoader$1 = new scriptLoader.ScriptLoader(commonOptions, resourcePath);
                        const result = scriptLoader$1.excute(scriptSetup, true);
                        code = code.replace(scriptSetup, result);
                    }
                    return { code };
                }
                else {
                    return { code };
                }
            }
            catch (err) {
                console.error("vue3-i18n-plugin error:", err);
                return null;
            }
        },
    };
}

async function loaderForWebpack(content) {
    const callBack = this.async();
    try {
        const filePath = this.resourcePath;
        const commonOptions = await scriptLoader.fileOperator.getConfig();
        if (!scriptLoader.validateFileType(filePath, commonOptions)) {
            return callBack(null, content);
        }
        const resourcePath = path.normalize(filePath);
        const localeFile = path.normalize(path.join(commonOptions.outputDir, `${commonOptions.anchorName}.js`));
        await scriptLoader.fileOperator.initMessage(localeFile);
        if ([".js", ".ts", ".cjs"].includes(path.extname(resourcePath))) {
            const scriptLoader$1 = new scriptLoader.ScriptLoader(commonOptions, resourcePath);
            const result = scriptLoader$1.excute(content, false);
            return callBack(null, result);
        }
        else if (filePath.endsWith(".vue")) {
            content = content.replace(/(<template[^>]*>)((.)*)<\/template>/gims, (match) => {
                const templateLoader = new scriptLoader.TemplateLoader(commonOptions, resourcePath);
                const result = templateLoader.excute(match);
                return `${result}`;
            });
            content = content.replace(/(<script[^>]*>)((.)*)<\/script>/gims, (_, preTag, content) => {
                const scriptLoader$1 = new scriptLoader.ScriptLoader(commonOptions, resourcePath);
                const result = scriptLoader$1.excute(content, false);
                return `${preTag}${result}<\/script>`;
            });
            return callBack(null, content);
        }
        console.log(filePath, 'filePath')
        if(filePath.includes('taskList\\taskList')){
            fs.writeFileSync('C:\\Users\\lenovo\\Desktop\\test.vue', content);
        }
        return callBack(null, content);
    }
    catch (error) {
        console.error(error);
        return callBack(error, content);
    }
}

const AIvueI18nPluginForVite = vueI18nPlugin;
const AIvueI18nLoaderForWebpack = loaderForWebpack;

exports.AIvueI18nLoaderForWebpack = AIvueI18nLoaderForWebpack;
exports.AIvueI18nPluginForVite = AIvueI18nPluginForVite;
exports.default = AIvueI18nLoaderForWebpack;
